operator/src/main/java/org/bf2/operator/managers/StrimziManager.java

```diff
@@ -128,12 +130,31 @@ public void onDelete(Deployment deployment, boolean deletedFinalStateUnknown) {
                             log.debugf("Delete event received for Deployment %s/%s",
                                     deployment.getMetadata().getNamespace(), deployment.getMetadata().getName());
                             deleteStrimziVersion(deployment);
+                            deleteLeadershipLeaseIfPresent(deployment);
                             updateStatus();
                         }
                     }
                 });
     }
 
+    void deleteLeadershipLeaseIfPresent(final Deployment deployment) {
+        final String leaseName = deployment.getMetadata().getName() + "-leadership-token";
```

Comment:
```suggestion
        // See https://github.com/fabric8io/kubernetes-client/issues/4638
        final String leaseName = deployment.getMetadata().getName() + "-leadership-token";
```

Reply:
Added it as javadoc to the method, definitely worth a comment for this oddity